pages/[username].js

```diff
@@ -24,14 +20,12 @@ export async function getServerSideProps(context) {
   log = logger.child({ username: username, ip: requestIp.getClientIp(req) });
   let data = {};
 
-  try {
-    const resUser = await fetch(
-      `${process.env.NEXT_PUBLIC_BASE_URL}/api/users/${username}`
-    );
-    data = await resUser.json();
+  const { status, userData } = await getUserApi(username);
+  if(status === 200) {
+    data = userData;
     log.info(`data loaded for username: ${username}`);
-  } catch (e) {
-    log.error(e, `profile loading failed for username: ${username}`);
+  } else {
+    log.error(userData.error, `profile loading failed for username: ${username}`);
   }
 
   if (!data.username) {
```

Comment:
I wonder if this could be moved to also remove the `else`

```js
if (!data.username || status !== 200) {
```

Reply:
If  username is not found, status will not be 200.
So I combined the fail paths into one.